server/server.go

```diff
@@ -0,0 +1,67 @@
+package server
+
+import (
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type ServerOptions struct {
+	Token     string
+	BaseURL   string
+	PublicURL string
+	Bind      string
+	Port      int
+}
+
+type Server struct {
+	status ServerStatus
+}
+
+func (self *Server) Start(options *ServerOptions) error {
+	self.reload()
```

Comment:
Usually the projects I have seen in go don't use the "self" or "this" pattern. Usually I have seen people using only the first letter of the type. In this case `s`.